### PR TITLE
Stop CupertinoScrollbar's track from paging the scroll view on tap

### LINKED ISSUE #120429: Disable tap on track of iOS scrollbar
## Steps to Reproduce

1. Create a scrolling widget with a Scrollbar

**Expected results:** 
A scrollbar that behaves exactly like iOS scrollbar. It's close, but if you tap the track of the scrollbar (so below or above the thumb), it scrolls a full page. This is not like native

**Actual results:** 
It is always interactive. This is not the case on the native scrollbar in iOS. If you tap the track on native iOS scrollbar, nothing happens

<details>
<summary>Code sample</summary>
easy
```dart
Scrollbar(child: SingleChildScrollView(
child: SizedBox(height: 1000))
```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 3.7.1, on macOS 13.1 22C65 darwin-arm64, locale nl-NL)
    • Flutter version 3.7.1 on channel stable at /Users/wouter/Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (9 days ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/wouter/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_SDK_ROOT = /Users/wouter/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.75.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

```

</details>

### PR DESCRIPTION
**Pull request by victorsanni**
Fixes https://github.com/flutter/flutter/issues/120429

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by MitchellGoodwin — COMMENTED** [review]
I'm not sure, but we probably should scroll on tap if on MacOS? cc @cbracken

**Comment by Piinks** [comment]
> I'm not sure, but we probably should scroll on tap if on MacOS?

Yep! We definitely should to match native behavior.

**Comment by cbracken** [comment]
Summary from chat is that the default on macOS is to scroll up/down a page, but in the Appearance system settings, you can set that to scroll to clicked position.

**Comment by victorsanni** [comment]
That's true, macOS does scroll on tap. But iOS doesn't? So should it be conditional based on if macOS vs. iOS? Using defaultTargetPlatform?


https://github.com/user-attachments/assets/20e92a0a-7485-4c6a-83b4-750955ecf184


https://github.com/user-attachments/assets/ff47c528-8eab-4db3-918c-c313b03658e6

**Comment by Piinks** [comment]
I think defaultTargetPlatform is used to do similar specific alterations like this in RawScrollbar

**Comment by victorsanni** [comment]
> I think defaultTargetPlatform is used to do similar specific alterations like this in RawScrollbar

I wonder if this feature should be baked into all `RawScrollbar`s on iOS or only `CupertinoScrollbar`. I remember reading a doc on the types of adaptivity: https://docs.google.com/document/d/14mxhVeVk1lClPkBi6VmB260_7fKUlZTM-Vn1bOres7M/edit. Specifically, I wonder if this paging on tap behavior falls into category 1 (low-level OS behavior) or category 4 (leaf-widgets with 1-1 analog) as described in the doc.

**Comment by MitchellGoodwin** [comment]
Imo, it should be put into `CupertinoScrollbar`. This is less a physics difference that should never change, and more a behavior. Which, according to Chris' comment, we will eventually want to add in conditionally scrolling to the next page on MacOS, which it makes more sense to configure that in Cupertino.

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1799,7 +1799,11 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     _cachedController = null;
   }
 
-  void _handleTrackTapDown(TapDownDetails details) {
+  /// Handler called when the track is tapped in order to scroll to the tapped
+  /// position.
+  ///
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1804]:
```suggestion
```

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 2]
```diff
@@ -1799,7 +1799,11 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     _cachedController = null;
   }
 
-  void _handleTrackTapDown(TapDownDetails details) {
+  /// Handler called when the track is tapped in order to scroll to the tapped
+  /// position.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1803]: I think so, right? Or does it jump to that specific position now?
```suggestion
  /// Handler called when the track is tapped in order to page in the tapped direction
  /// position.
```

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 3]
```diff
@@ -1799,7 +1799,11 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
...
+  ///
+  @protected
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1805]: We should add `@mustCallSuper` like the other public handlers

File: packages/flutter/test/cupertino/scrollbar_test.dart [hunk 4]
```diff
@@ -1075,7 +1075,64 @@ void main() {
         rrect: RRect.fromLTRBR(794.0, 3.0, 797.0, 359.4, const Radius.circular(1.5)),
       ),
     );
-  });
+  }, variant: TargetPlatformVariant.all(excluding: <TargetPlatform>{TargetPlatform.iOS}));
```
Comment (Piinks) [inline packages/flutter/test/cupertino/scrollbar_test.dart:1078]: TIL about `excluding`! Cool!

File: packages/flutter/lib/src/widgets/scrollbar.dart (hunk 2 above)
Comment (victorsanni) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1803]: It pages, not scrolls. Thanks for the catch.

**Review by Piinks — APPROVED** [review]
LGTM!

**Review by MitchellGoodwin — APPROVED** [review]
LGTM!

**Pull request merged**